### PR TITLE
Add v-html attribute rendering

### PR DESCRIPTION
**Pull request by JonasKress**
Allow to add HTML to node via v-html attribute.
Test still missing...

### CODE REVIEW

File: src/Component.php [hunk 1]
```diff
@@ -241,6 +243,35 @@ private function handleFor( DOMNode $node, array $data ) {
...
+			$variableName = $node->getAttribute( 'v-html' );
+			$node->removeAttribute( 'v-html' );
+
+			$newNode = $node->cloneNode( true );
+
+			$this->appendHTML( $newNode, $data[$variableName] );
+
+			$node->parentNode->insertBefore( $newNode, $node );
+
+			$this->removeNode( $node );
```
Comment (thiemowmde) [inline src/Component.php]: This looks like it should be `$node->parentNode->replaceChild( $newNode, $node )`

File: src/Component.php [hunk 2]
```diff
@@ -241,6 +243,35 @@ private function handleFor( DOMNode $node, array $data ) {
...
+		$tmpDoc = new DOMDocument();
+		$tmpDoc->loadHTML( $source );
+		foreach ( $tmpDoc->getElementsByTagName('body')->item(0)->childNodes as $node ) {
+			$node = $parent->ownerDocument->importNode( $node, true );
+			$parent->appendChild( $node );
+		}
+	}
+
+	private function handleRawHtml( DOMNode $node, array $data ) {
+		if ( $this->isTextNode( $node ) ) {
```
Comment (thiemowmde) [inline src/Component.php]: This looks like you should do an `instanceof DOMElement` check instead. What if there is a third type that is neither text nor an element? The current approach must fail then.

File: src/Component.php [hunk 3]
```diff
@@ -241,6 +243,35 @@ private function handleFor( DOMNode $node, array $data ) {
 		}
 	}
 
+	private function appendHTML( DOMNode $parent, $source ) {
+		$tmpDoc = new DOMDocument();
+		$tmpDoc->loadHTML( $source );
+		foreach ( $tmpDoc->getElementsByTagName('body')->item(0)->childNodes as $node ) {
```
Comment (thiemowmde) [inline src/Component.php]: This assumes there is always a body. Is this guaranteed? Honest question – I don't know.

File: src/Component.php (hunk 2 above)
Comment (manicki) [reply src/Component.php]: I guess you might be right. It might be worth addressing separately IMO, as similar check is used in other places too.

**Comment by JonasKress** [comment]
+2

**Pull request merged**